src/odemis/util/angleres.py

```diff
@@ -87,76 +84,114 @@ def _extractAngleInformation(data, hole, dtype=None):
-    xpix = mirror_x - jj  # x coordinates of the pixels
-
-    # populate arrays with values
-    for i in xrange(image_x):
-        ypix = (i - mirror_y) + (2 * parabola_f) / pixel_size[1]  # y coordinates of the pixels
-
-        # TODO can we modify xpix and ypix so we only search for angles within half circle
-        # Finds the angles of emission by the sample for each px (x,y) in the raw data.
-        # These angles only depend on the mirror geometry.
-        # Each px in the raw data corresponds to a specific theta-phi combination
-        theta, phi, omega = _FindAngle(data, xpix, ypix, pixel_size)
-
-        # theta_data: array containing theta values for each px in raw data
-        theta_data[i, :] = theta
-        # phi_data: array containing phi values for each px in raw data
-        phi_data[i, :] = phi
-
-        # intensity_data contains the intensity values from raw data.
-        # It already reflects the shape of the mirror
-        # and is normalized by omega (solid angle:
-        # measure for photon collection efficiency depending on theta and phi)
-        intensity_data[i, :] = cropped_image[i] / omega
+    x_indices = numpy.linspace(0, data.shape[1] - 1, data.shape[1])  # list of px indices
+    # correct x coordinates to be 0 at the hole position
+    x_pos = x_indices - pole_pos[0]  # x coordinates of the pixels (horizontal)
+
+    y_indices = numpy.linspace(0, data.shape[0] - 1, data.shape[0])  # list of px indices
+    # correct y coordinates to be 0 at the hole position
+    y_pos = (y_indices - pole_pos[1]) + (2 * parabola_f) / pixel_size[0]  # y coordinates of the pixels (vertical)
+
+    # create two arrays with set of x and y coordinates and with same shape of data
+    x_array, y_array = numpy.meshgrid(x_pos, y_pos)
+
+    # TODO can we modify xpos and ypos so we only search for angles within half circle
+    # Finds the angles of emission by the sample for each px (x,y) in the raw data.
+    # These angles only depend on the mirror geometry.
+    # Each px in the raw data corresponds to a specific theta-phi combination
+    # theta_data: array containing theta values for each px in raw data
+    # phi_data: array containing phi values for each px in raw data
+    theta_data, phi_data, omega = _FindAngle(x_array, y_array, pixel_size, parabola_f)
+
+    # intensity_data contains the intensity values from raw data.
+    # It already reflects the shape of the mirror
+    # and is normalized by omega (solid angle:
+    # measure for photon collection efficiency depending on theta and phi)
+    intensity_data = cropped_image / omega
 
     return theta_data, phi_data, intensity_data, circle_mask_dilated
 
 
-def AngleResolved2Polar(data, output_size, hole=True, dtype=None):
+def _FindAngle(x_array, y_array, pixel_size, parabola_f):
+    """
+    For given pixels, finds the angle of the corresponding ray
+    :parameter x_array: (ndarray) x coordinates of the pixels
```

Comment:
Please add the expected dimension of the ndarrays. They are all a 2D array, right?